Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLParameterScan.py

```diff
@@ -0,0 +1,277 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.api import *
+from mantid.kernel import *
+from mantid.simpleapi import *
+from SANSILLAutoProcess import needs_loading, needs_processing
+
+
+class SANSILLParameterScan(DataProcessorAlgorithm):
+    """
+    Performs treatment for scans along a parameter for D16.
+    """
+    progress = None
+    reduction_type = None
+    sample = None
+    absorber = None
+    container = None
+    sensitivity = None
+    mask = None
+    default_mask = None
+    output = None
+    output_sens = None
+    normalise = None
+    thickness = None
+    output2D = None
+    observable = None
+    pixel_y_min = None
+    pixel_y_max = None
+
+    def category(self):
+        return 'ILL\\SANS;ILL\\Auto'
+
+    def summary(self):
+        return 'Integrate SANS scan data along a parameter'
+
+    def seeAlso(self):
+        return []
+
+    def name(self):
+        return 'SANSILLParameterScan'
+
+    def setUp(self):
+        self.sample = self.getPropertyValue('SampleRuns')
+        self.absorber = self.getPropertyValue('AbsorberRuns')
+        self.container = self.getPropertyValue('ContainerRuns')
+        self.sensitivity = self.getPropertyValue('SensitivityMaps')
```

Comment:
This algorithm is for one detector configuration, so there should be just one sensitivity map, only one mask, and absorber and container can only be summed correspondingly, but not listed.